app/benchmark.go

```diff
@@ -0,0 +1,278 @@
+	if blockTimeCount <= 0 {
+		return 0
+	}
+	avgBlockTime := totalBlockTime / time.Duration(blockTimeCount)
+	return avgBlockTime.Milliseconds()
+}
+
+// flushStats holds the statistics for a flush window
+type flushStats struct {
+	txCount        int64
+	blockCount     int64
+	latestHeight   int64
+	maxBlockTimeMs int64
+	avgBlockTimeMs int64
+	tps            float64
+	peakTps        float64
+}
+
+// getAndResetStats atomically reads current stats and resets counters for next window
+func (l *benchmarkLogger) getAndResetStats(now time.Time) (flushStats, time.Time) {
+	l.mx.Lock()
+	defer l.mx.Unlock()
+
+	stats := flushStats{
+		txCount:        l.txCount,
+		blockCount:     l.blockCount,
+		latestHeight:   l.latestHeight,
+		maxBlockTimeMs: l.maxBlockTime.Milliseconds(),
+	}
+
+	prevTime := l.lastFlushTime
+	totalBlockTime := l.totalBlockTime
+	blockTimeCount := l.blockTimeCount
+
+	// Reset counters for next window (but keep prevBlockTime and peakTps for continuity)
+	l.txCount = 0
+	l.blockCount = 0
+	l.latestHeight = 0
+	l.maxBlockTime = 0
+	l.totalBlockTime = 0
+	l.blockTimeCount = 0
+	l.lastFlushTime = now
+
+	// Calculate TPS
+	duration := now.Sub(prevTime)
+	if duration > 0 && !prevTime.IsZero() {
+		stats.tps = calculateTPS(stats.txCount, duration)
+	}
+
+	// Calculate average block time
+	stats.avgBlockTimeMs = calculateAvgBlockTime(totalBlockTime, blockTimeCount)
+
+	// Update peak TPS if current TPS is higher
+	if stats.tps > l.peakTps {
+		l.peakTps = stats.tps
+	}
+	stats.peakTps = l.peakTps
+
+	return stats, prevTime
+}
+
+func (l *benchmarkLogger) FlushLog() {
+	now := time.Now()
+	stats, _ := l.getAndResetStats(now)
+
+	l.logger.Info("benchmark",
+		"txs", stats.txCount,
+		"blocks", stats.blockCount,
+		"height", stats.latestHeight,
+		"blockTimeMax", stats.maxBlockTimeMs,
+		"blockTimeAvg", stats.avgBlockTimeMs,
+		"tps", stats.tps,
+		"peakTps", stats.peakTps,
+	)
+}
+
+func (l *benchmarkLogger) Start(ctx context.Context) {
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			l.FlushLog()
+		}
+	}
+}
+
+func NewGeneratorCh(ctx context.Context, txConfig client.TxConfig, chainID string, evmChainID int64, logger log.Logger) <-chan *abci.ResponsePrepareProposal {
+	gen, err := generator.NewConfigBasedGenerator(&config.LoadConfig{
+		ChainID:    evmChainID,
+		SeiChainID: chainID,
+		Accounts:   &config.AccountConfig{Accounts: 5000},
+		Scenarios: []config.Scenario{{
+			Name:   scenarios.EVMTransfer,
+			Weight: 1,
+		}},
+	})
+	if err != nil {
+		panic("failed to initialize generator: " + err.Error())
+	}
+	ch := make(chan *abci.ResponsePrepareProposal, 100)
+	go func() {
+		defer close(ch)
+		var height int64
+		for {
+			// bail on ctx err
+			if ctx.Err() != nil {
+				return
+			}
+			// generate txs like: txs := gen.GenerateN(1000)
+			loadTxs := gen.GenerateN(1000)
```

Comment:
As a side note: wouldn't it make more sense to pull sei-load into sei-chain monorepo if we're going to be using it in this way within monorepo as opposed to a separately maintained codebase?

Reply:
Yeah I think we should - I'm thinking we should do that in an isolated PR just so it's a noop change that's easier to review